src/kbmod/result_list.py

```diff
@@ -890,6 +890,20 @@ def revert_filter(self, label=None):
 
         return self
 
+    def sync_table_indices(self, table):
+        """Syncs the entries in the current list with those in a table version
+        of the results by filtering on the 'index' column. Rows that do not
+        appear in the table are removed from the list. The indices in the table
+        are updated to match the new ordering in the result list.
+
+        Parameters
+        ----------
+        table : `astropy.table.Table`
+            A table with the data as generated by to_table().
+        """
+        self.filter_results(table["index"], "Table filtered")
+        table["index"] = range(len(table))
```

Comment:
Just a comment for when we actually address https://github.com/dirac-institute/kbmod/issues/535 : I'm presuming we want to actually set an [astropy.table index](https://docs.astropy.org/en/stable/table/indexing.html#creating-an-index), but curious if there are any specific reasons why we were choosing not to do so (for example extra performance cost not worth it for hopefully small ResultLists)